sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/Identity.java

```diff
@@ -380,6 +387,16 @@ public void run() {
         });
     }
 
+    private void instanceSignOut() {
+        post(new Runnable() {
+
+            @Override
+            public void run() {
+                mTokenStorage.removeToken();
```

Comment:
Also, we need to synchronize it to avoid thread issues

Reply:
Actually not needed here as it's always accessed from the background looper.

We usually don't add synchronized to things accessed solely on background thread, especially after https://github.com/Microsoft/AppCenter-SDK-Android/pull/665 happened.

The code is not harmonized.

Reply:
@guperrot `mSignInDelayed` is accessed from UI thread too. So, we need to synchronize it.